src/schema/sources.ts

```diff
@@ -1481,6 +1539,67 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         true,
       );
     },
+    sourceAwards: async (
+      _,
+      args: GQLCommentAwardArgs,
+      ctx: Context,
+      info,
+    ): Promise<Connection<GQLUserAward>> => {
+      await ensureSourcePermissions(ctx, args.id);
+      await ctx.con.getRepository(Source).findOneOrFail({
+        select: {
+          id: true,
+        },
+        where: { id: args.id },
+      });
+
+      const pageGenerator = offsetPageGenerator<GQLUserAward>(20, 100);
+      const page = pageGenerator.connArgsToPage(args);
+
+      return graphorm.queryPaginated(
+        ctx,
+        info,
+        (nodeSize) => pageGenerator.hasPreviousPage(page, nodeSize),
+        (nodeSize) => pageGenerator.hasNextPage(page, nodeSize),
+        (node, index) => pageGenerator.nodeToCursor(page, args, node, index),
+        (builder) => {
+          builder.queryBuilder.andWhere(
+            `${builder.alias}.flags->>'sourceId' = :sourceId`,
+            {
+              sourceId: args.id,
+            },
+          );
+
+          builder.queryBuilder.limit(page.limit).offset(page.offset);
+
+          return builder;
+        },
+        undefined,
+        true,
+      );
+    },
+    sourceAwardsTotal: async (
+      _,
+      args: GQLCommentAwardArgs,
+      ctx: Context,
+    ): Promise<{ amount: number }> => {
+      await ensureSourcePermissions(ctx, args.id);
+      await ctx.con.getRepository(Source).findOneOrFail({
```

Comment:
Same her as above comment, this findOneOrFail is redundant as `ensureSourcePermissions` already does the same.